src/main/java/fi/aalto/cs/apluscourses/utils/APlusLocalizationUtil.java

```diff
@@ -3,20 +3,27 @@
 import java.util.Objects;
 import org.apache.commons.lang3.StringUtils;
 import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
 
 public class APlusLocalizationUtil {
 
+  private static String extractLocalizedText(@NotNull String localizedString, @NotNull String languageCode) {
+    return StringUtils.substringBetween(localizedString, "|" + languageCode + ":", "|");
+  }
+
   /**
-   * Parses the English name from the given string, or returns the string if the string contains
-   * no localization. An example of a valid string is {@code "|en:hello|fi:terve|"}.
+   * Parses the given localized string and returns the text corresponding to the specified language.
+   * If the entry in the requested language does not exist, the method attempts to localize the string in English.
+   * If that fails too, the whole raw string is returned.
```

Comment:
Although not important right now, I'd find it more intuitive that if the string has the localized format (with vertical bars and colons), it's not returned as-is, but some language variant (say, the first) is returned as a backup plan if there's neither the requested nor English available. The raw string would be only returned if there was no vertical bars and  colons at all.